code/datums/components/storage/concrete/egun_slot.dm

```diff
@@ -0,0 +1,19 @@
+/datum/component/storage/concrete/egun_slot/handle_item_insertion(obj/item/W, prevent_warning = FALSE, mob/living/user)
+	var/atom/P = src.parent
+	var/list/things = contents()
+	var/total_guns = 0
+	var/max_guns = 1 //max guns allowed
+
+	if(istype(W, /obj/item/gun/energy))
+		for(var/i in things)
+			var/obj/item/E = i
+			if(istype(E, /obj/item/gun/energy))
+				total_guns++
```

Comment:
this doesn't make sense, why are you checking for gun 2 times?

Reply:
added comments to explain